extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java

```diff
@@ -564,6 +564,11 @@ public T call(HttpServerExchange exchange, C context) throws Exception {
                         // Not sure what to do here
                         ManagedContext requestContext = beanContainer.requestContext();
                         if (requestContext.isActive()) {
+                            if (currentVertxRequest.getCurrent() == null && exchange != null
```

Comment:
Can you add a comment about what is actually causing this condition? (i.e. what is activating the request scope without also setting the current request).

This seems fine, but I would like to understand why it is happening.

Reply:
I don't really understand how @ActivateRequestContext in the augmentor could do this. It should be terminated after the invocation: https://github.com/stuartwdouglas/quarkus/blob/8398d79f770b7f5b7e1cb6c90397072cafee6264/independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ActivateRequestContextInterceptor.java#L131

Reply:
Request context activated in the augmentor is terminated as you say, however when `io.quarkus.vertx.http.runtime.security.HttpAuthorizer#doPermissionCheck` is run, context snapshot is created from `SlowActiveContextState` (previously restored by `SlowCapturedContextState`) and at the moment when `ServletInitialHandler` handles request the state is active again. It's done by Smallrye Context Propagation and I can't tell whether this is expected.

I have limits in understanding of above-mentioned area, I can look deeper to context propagation on Thursday evening (or do required changes then), but if someone could give me a hint on expected behavior of context propagation I'd be grateful.

Reply:
btw. you know it's down to the `@ActivateRequestContext` as if you comment it out, problem disappear.

Reply:
Can't tell based on https://github.com/smallrye/smallrye-context-propagation whether this is expected behavior, but @FroMage @mkouba are maintainers of Smallrye Context Propagation extension. Could you kindly give me a hint whether what I describe in the comment above is expected, please?

Reply:
> io.quarkus.vertx.http.runtime.security.HttpAuthorizer#doPermissionCheck is run, context snapshot is created from SlowActiveContextState 

SR CP just propagates (by default) all contexts when asked to - for CDI this means req. scope if it was active. So I guess the question is whether it's expected that someone activates the context at that point? Also note that the `ActivateRequestContextInterceptor` can still terminate the req. context on the original thread (in terms of its state - the beans it had) and so any thread that CP propagates it into would then seem to have req, context active but would in fact be a fresh instance of req. context.

Reply:
> SR CP just propagates (by default) all contexts when asked to - for CDI this means req. scope if it was active. So I guess the question is whether it's expected that someone activates the context at that point?

reproducer is more or less c&p of docs example https://quarkus.io/guides/security-customization#security-identity-customization including request context activation, so IMO yes

> Also note that the `ActivateRequestContextInterceptor` can still terminate the req. context on the original thread (in terms of its state - the beans it had) and so any thread that CP propagates it into would then seem to have req, context active but would in fact be a fresh instance of req. context.

@manovotn that's exactly what happens, thank you for taking your time to answer

Reply:
@stuartwdouglas I added comment as you suggested, should the comment be improved, just let me know.